From 412fcf217eafe67b5db992b87aac2bbfbb1438ad Mon Sep 17 00:00:00 2001
From: "qipeng.zha" <qipeng.zha@intel.com>
Date: Wed, 6 May 2015 23:57:39 +0800
Subject: [PATCH] Touch: update firmware only when IDs are same

There exist different versions of touch controller on
same build, so add this control to avoid update wrong
touch firmware and config

Change-Id: I13ec10383a72bd94e45c5249e42f82f71ae22a63
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-8775
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1562
Signed-off-by: qipeng.zha <qipeng.zha@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index b47c5e72fcb5..7a055830f586 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -2614,15 +2614,15 @@ static bool mxt_fw_is_latest(struct mxt_data *data, struct mxt_info *info,
 		return false;
 	}
 
+	/* update firmware when fw version is different and IDs are same */
 	if (fw_info->family_id == info->family_id &&
-		fw_info->variant_id == info->variant_id &&
-		fw_info->version == info->version &&
-		fw_info->build == info->build)
-		return true;
-	else {
+			fw_info->variant_id == info->variant_id &&
+			(fw_info->version != info->version ||
+			fw_info->build != info->build)) {
 		dev_info(&data->client->dev, "Update firmware\n");
 		return false;
-	}
+	} else
+		return true;
 }
 
 static int mxt_check_firmware(struct mxt_data *data, const struct firmware *fw)
-- 
1.9.1

